clients/src/test/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumerTest.java

```diff
@@ -2077,6 +2077,19 @@ void testReaperInvokedInPoll() {
         verify(backgroundEventReaper).reap(time.milliseconds());
     }
 
+    @Test
+    public void testUnsubscribeWithoutGroupId() {
+        consumer = newConsumerWithoutGroupId();
+        completeFetchedCommittedOffsetApplicationEventExceptionally(new TimeoutException());
+        doReturn(Fetch.empty()).when(fetchCollector).collectFetch(any(FetchBuffer.class));
```

Comment:
do we need this? I wouldn't expect we have to mock anything related to committed offsets or fetching because we're not polling in the test (or using committed offsets in any way) right?

Reply:
Yeah, removed unused mock. Thanks for the review. 👍